src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -61,8 +61,12 @@ void UpdatePreReleaseTag(GitVersionContext context, SemanticVersion semanticVers
             if (tagToUse == "useBranchName")
             {
                 Logger.WriteInfo("Using branch name to calculate version tag");
-                var name = branchNameOverride ?? context.CurrentBranch.Name;
-                tagToUse = name.RegexReplace(context.Configuration.BranchPrefixToTrim, string.Empty, RegexOptions.IgnoreCase);
+                tagToUse = branchNameOverride ?? context.CurrentBranch.Name;
+                if (!string.IsNullOrWhiteSpace(context.Configuration.BranchPrefixToTrim))
```

Comment:
I could have made my change to work on the `name` variable that was used, but since a regex invoke when this often is an empty string I found that this perhaps should be optimized at the same time. So that the regex replace is done only when needed. But, would work without it I guess.